tempesta_fw/ptrset.h

```diff
@@ -0,0 +1,296 @@
+/**
+ *		Tempesta FW
+ *
+ * The TfwPtrSet is a generic set of pointers. Essentially, this is just a small
+ * array of pointers, where all pointers are unique and you can add/delete
+ * elements and iterate over them.
+ *
+ * The purpose of a TfwPtrSet is to store a small limited set of pointers to
+ * distinct objects like a list of back-end servers for the Tempesta FW.
+ *
+ * Advantages over a linked list are:
+ *  - No need for augmenting the target data structure with sturct list_head.
+ *    This is useful when you want to make a list of objects in an external
+ *    kernel module, but you don't want the main code to know about your module.
+ *  - Lightweight operations.
+ *    Although the asymptotic behavior is bad (O(N) for most operations), the
+ *    code is good at constant factors, so the performance should be slightly
+ *    better for small number of elements.
```

Comment:
It's hard to say whether the pointers are the right solution w/o any application in the code. The last factor says that they should not be at least in the patch set.

The fadventages are doubtful - we usually use intrusive lists to get structures stored in the list in the cache while we're traversing the list, so we get the hot data when we find the required item.